woocommerce/woocommerce
# Update PR template and auto change log with better no-entry support

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR updates the PR template and the auto change log utility to add support for skipping a changelog entry via a separate checkbox. A comment is still required if you opt to skip a changelog entry. It also adds support for specifying a separate GH token for the creation of the change file commit, which allows for our CI to be run on that commit.





Closes #43431 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a fork of this repository if you do not already have one.
2. If actions are enabled, disable the "Changelog Auto Add" workflow.
3. Merge the changes from this PR to trunk of your fork.
4. Create a PR in your fork (targeting trunk of your fork) affecting a project that uses the changelogger (such as WooCommerce core). Do not create a change file. Note the PR number.
5. Update the PR description to check "This Pull Request does not require a changelog entry.", and add a comment.
6. Run `GITHUB_TOKEN=... pnpm utils changefile -o jonathansadowski 3`. Replace `jonathansadowski` with the owner of your fork. Replace `3` with your PR number, and be sure to specify your `GITHUB_TOKEN`.
7. Verify that a new commit is added to your PR containing a changefile with no entry and with a comment.
8. Re-enable actions in your fork, and be sure to enable the "Changelog Auto Add" workflow.
9. Optionally: Add a secret within your repo with the name `PR_CREATE_TOKEN`
10. Repeat the previous steps, but instead of running the utility locally, make sure that the workflow runs and adds the commit. If you added a token it will come from that user, otherwise it will come from the github bot.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
